glmmbernoulli.qmd

```diff
@@ -0,0 +1,523 @@
+Comparisons of model fits through different software systems, as provided by the Center for Multilevel Modelling, incorporate only a linear term in age, even though the pattern is clearly nonlinear.
+The lesson here is similar to what we have seen in other examples; careful plotting of the data should, whenever possible, precede attempts to fit models to the data.
+
+### Initial GLMM fit to the contraception data {#sec-contraglmm}
+
+Fitting a generalized linear mixed model (GLMM) is very similar to fitting a linear mixed model (LMM).
+We call the `fit` function with the first three arguments being the model type, `MixedModel`, then a formula specifying the response, fixed-effects terms and random-effects terms, then a data table.
+For GLMMs we also specify a fourth positional argument which is a distribution - in this case `Bernoulli()`.
+
+Establishing the contrasts and fitting a preliminary model with random effects for `dist`, main effects for `livch`, `age`, `age^2`, and `urban`, plus interaction terms for `age & urban` and `age^2 & urban` can be done as
+
+```{julia}
+contrasts[:livch] = EffectsCoding(; base="0")
+contrasts[:urban] = HelmertCoding()
+contrasts[:dist] = Grouping()
+
+com01 = let
+  form = @formula(use ~ 1 + livch + age * age * urban + (1 | dist))
+  fit(MixedModel, form, contra, Bernoulli(); contrasts, nAGQ)
+end
+```
+
+Notice that in the formula language defined by the [StatsModels](https://github.com/JuliaStats/StatsModels.jl) package, an interaction term is written with the `&` operator.
+*Crossing* of terms, which generates main effects and interactions, is written with the `*` operator (as in the formula language in [R](https://r-project.org)).
+An interaction of a numeric variable with itself is performed by multiplication so the coefficient labelled `age & age` in the table is the quadratic term in `age`.
+(Notice that, in this formula language, `age * age`, which may easily be interpreted as `age^2`, expands to `age + age^2`.)
+Thus, what is written in the formula as a three-way interaction `age * age * urban` becomes an `urban` contrast plus linear and quadratic terms for `age` and each of their interactions with the `urban` contrast, which will be coded as `-1` for `N` and `+1` for `Y`.
+
+A fifth positional argument can be used to specify the link function, described in @sec-glmmlink, but in most cases the canonical link function for the distribution is used.
+In the case of the Bernoulli distribution the canonical link is the *logit* link.
+
+As for LMMs, the named argument `contrasts` specifies the contrasts to apply to some of the covariates in a key-value dictionary.
+Another named argument, `nAGQ`, specifies the number of quadrature points to use in an *adaptive Gauss-Hermite quadrature* rule for evaluating the deviance (see @sec-aGHQ for details).
+A small, odd number, such as `nAGQ=9` defined in the first code block of this chapter, is the preferred choice.
```

Comment:
Does it make sense to distinguish between scalar and vector RE here? Or how the benefit of additional quadrature points falls off rapidly?

(don't know if you've discussed this further below or in the nAGQ appendix; I'm commenting as I read.)